### ISSUE TITLE
Add an official project generator script / project skeleton

### ISSUE THREAD

**Issue by pdesaulniers** [DISTRHO/DPF#54]
As we've previously discussed. 

DPF plugins use similar project structures and Makefiles, so there should probably be an official way of starting a DPF project.

For those who already want something like that, this project is pretty neat: https://github.com/SpotlightKid/cookiecutter-dpf-effect

**Comment by codecat** [comment]
Perhaps use something like CMake/PreMake/GENie to build DPF, so that projects can be generated for whatever platform. If implemented well, you could make plugins that use DPF pretty easily by including a project file directly from DPF to link against.

**Comment by pdesaulniers** [comment]
The cookiecutter project has been updated to support UI generation and MIDI In/Out. For the time being, I think it's the sanest way of starting a new DPF project.

@codecat I'm not super familiar with these build systems. Can they handle something similar to [the jinja templates we're using in the cookiecutter project](https://github.com/SpotlightKid/cookiecutter-dpf-effect/blob/master/%7B%7B%20cookiecutter.repo_name%20%7D%7D/plugins/%7B%7B%20cookiecutter.plugin_name%20%7D%7D/%7B%25-%20if%20cookiecutter.ui_type%20!%3D%20%22none%22%20%25%7DUI%7B%7B%20cookiecutter.plugin_name%20%7D%7D.cpp%7B%25-%20endif%20%25%7D)?

**Comment by codecat** [comment]
No, it doesn’t work like a project template system. It merely provides an easy way for cross-platform project generation. The source files will still need to manually be written.

**Comment by mxmilkb** [comment]
It would be handy if a template Makefile could have bits to uncomment for different setups, like compiling for LV2 only.

**Comment by pdesaulniers** [comment]
@mxmilkb The cookiecutter project has some flags in a Makefile for this: https://github.com/SpotlightKid/cookiecutter-dpf-effect/blob/master/%7B%7B%20cookiecutter.repo_name%20%7D%7D/plugins/%7B%7B%20cookiecutter.plugin_name%20%7D%7D/Makefile#L48-L59

**Comment by mxmilkiib** [comment]
A note as per IRC; an official turnkey template to fork, especially that builds with GH actions, like the JUCE and VCV ones available, would be a good way to help on-board developers.

https://github.com/sudara/pamplejuce

https://github.com/maxwellpollack/juce-plugin-ci

https://community.vcvrack.com/t/automated-building-and-releasing-plugins-on-github-with-github-actions/11364

https://www.kvraudio.com/forum/viewtopic.php?t=557214
